add hospitalizations to reichlab formatter, changes from #125, cleanup

dajmcdon opened the pull request:
- This should format the hospitalization forecasts correctly as well. 
- Many R checks fail, from previous updates. Most associated with corrections.
- Check that corrections are working in Hosp Production

dajmcdon committed:
add hospitalizations to reichlab formatter

dshemetov commented:
Didn't follow every line of code exactly, but switching over to this branch stitched the results together correctly in production.

EDIT: Read it over in more detail and it looks good.

dshemetov committed:
Update the evalcast dependency branch and
* few minor lint issues (unexplicit tidyeval vars)
* bump Roxygen version
* remove unused packages (MMWRweek, fs)
* add implicit dependencies (covidcast, rlang, tidyselect)
* try to fix tests (still broken)

dshemetov committed:
merge add-hosp

dshemetov committed:
Merge branch 'ds/update-zookeeper-deps2' into ds/update-zookeeper-deps

dshemetov committed:
Add kjytay back to codeowners

lcbrooks committed:
Update stale `animalia/man`

Run `devtools::document` on `animalia` package.

lcbrooks committed:
Revert CODEOWNERS to smaller set, reduce review spam

lcbrooks committed:
Fix mangled, buggy, stochastic, ntwk-ing state corrector test, clean

- Fix mangling: fix parts of the test that were older versions / incomplete and
  unintended to be run
- Fix stochasticity: inputs to rmultinom_wrapper depend on output of
  rbinom_wrapper; test rbinom_wrapper inputs and control its outputs to reliably
  test the inputs to rmultinom_wrapper
- Abort if assumption about default window size parity doesn't hold
- Use dummy data rather than API query; speed things up and don't keep hitting
  the API
- Remove unnecessary warning-capturing now that warning has been removed
- Remove debug print statements
- Remove `.env` arg to `mockr::with_mock` that doesn't / no longer
  actually/universally works to make the test run properly when run directly in
  interactive mode rather than via `testthat`; debugging this issue seems a
  little mysterious, with `get("mocked_wrapper_fn")` not producing the same
  results as `mocked_wrapper_fn`, so instead, just iterate with
  `testthat::test_file("<path to zookeeper package
  directory>/tests/testthat/test-state-corrector.R")`

dshemetov committed:
Declare stringr, readr dependencies
* remove covidcast from suggests (it's in Imports)

brookslogan commented on utilities/zookeeper/R/reichlab-formatter.R:
```diff
@@ -37,21 +38,47 @@ format_predictions_for_reichlab_submission <- function(predictions_cards,
+      quantile = round(.data$quantile, 3)) %>%
     filter(.data$signal == "deaths_incidence_num" |
+             .data$signal == "confirmed_admissions_covid_1d" |
              allowed_quantiles(.data$quantile, case_quants)) %>%
```
Quantile forecasts are filtered to the accepted quantile levels for cases, but maybe not for deaths and hospitalizations.

brookslogan replied:
Additionally, around here, if there is already a point prediction provided, it might not be recognized and another point forecast based on the median might be tacked on.

brookslogan replied:
Filed away in #126 and #128.  These are just validation issues, so it's fine to merge this without addressing them.  However, I believe #128 or a similar issue has caused issues in some forecasting scripts, where the baseline model outputting a point prediction when all the other forecasters did not actually broke some ensemble geo-reaggregation code.

brookslogan commented on utilities/zookeeper/tests/testthat/test-reichlab-formatter.R:
```diff
@@ -1,26 +1,41 @@
                    each = 5),
-      target_end_date = rep(as.Date(c("2020-01-09", "2020-01-16", "2020-01-23")), each = 5),
-      incidence_period = "epiweek"
+      target_end_date = rep(as.Date(c("2020-01-09", "2020-01-16", "2020-01-23", "2020-01-23")), each = 5),
```
Might need to be 2020-01-30 for the 4-wk-ahead?

brookslogan replied:
or 2020-01-06 for 4 day ahead?

brookslogan replied:
It appears that it should be the latter.  This highlights that the formatter is not validating the `target_end_date`s.

brookslogan replied:
Additionally, the formatter is not validating that the `geo_value`s meet its expectations; e.g., it could get state fips codes and attempt to convert them "from abbrevations" to fips codes, resulting in NAs.  The existing partial guard against this is the check for the `"predictions_cards"` class, but, e.g., `as.predictions_cards` looks like it may still allow a user to encounter this case.

brookslogan replied:
Addressed.  Validation issue noted in #127.

brookslogan commented on utilities/zookeeper/tests/testthat/test-reichlab-formatter.R:
```diff
@@ -1,26 +1,41 @@
+      signal = rep(c("confirmed_incidence_num", "confirmed_incidence_num",
+                     "deaths_incidence_num", "confirmed_admissions_covid_1d"),
+                   each = 5),
+      target_end_date = rep(as.Date(c("2020-01-09", "2020-01-16", "2020-01-23", "2020-01-23")), each = 5),
```
Again, here.  2020-01-23 vs 2020-01-30 for 4 wk ahead

brookslogan replied:
or 2020-01-06 for 4 day ahead?

brookslogan replied:
Should be the latter.

brookslogan replied:
Addressed.  Validation issue noted in #127.

brookslogan commented on utilities/zookeeper/tests/testthat/test-reichlab-formatter.R:
```diff
@@ -1,26 +1,41 @@
+    class(pcard1) <- c("predictions_cards", class(pcard1))
 
-    out <- format_predictions_for_reichlab_submission(pcard)
+    out <- format_predictions_for_reichlab_submission(pcard1)
```
pcard might not be used in this test

brookslogan replied:
--- stopped looking at this point.  will pick up from here later ---

brookslogan replied:
--> removing or commenting out definition of pcard is fine

brookslogan committed:
Merge pull request #125 from dshemetov/ds/update-zookeeper-deps

Update zookeeper dependency on evalcast; cleanup various zookeeper things, animalia DESCRIPTION

lcbrooks committed:
Clean extra object, bad `target_end_date`s in formatter tests

brookslogan commented:
This looks good to me.  Tests pass locally for me.  @dshemetov feel free to merge.  (Not sure if you can unassign yourself to review and re-approve, then merge?  Or if merging the PR #125 into this disqualified us from acting as reviewers.  Anyway, to consider this fully reviewed you might want to check that I didn't break the tests/build for you; my last commit here only touched the tests.)

dshemetov commented:
While I'm not a codeowner, you have the power to give an approving review and merge. How we get it to re-request a review from you though, I'm not sure. Maybe if you go to the file compare, you'll see an opportunity for another review up in the top right?

lcbrooks committed:
Fix missing man-roxygen entry in .Rbuildignore

The man-roxygen directory contains roxygen definitions for @template and
@templateVar, and should not be included in the package build. Run
`usethis::use_build_ignore("man-roxygen")` to add an .Rbuildignore entry for it.

lcbrooks committed:
Fix broken links, missing params, and text in wrapper docs

brookslogan commented:
I did a little additional cleanup in this PR (in addition to your cleanup inherited from #125) to satisfy some checks and improve some docs.

Re-request worked.  Not sure if I actually had to remove my own assignment first or not.  GitHub seems a little sassy or nervous with the "self-requested a review" but allows the merge.

dshemetov commented:
Nice thanks! Let's delete `add-hosp` too, to keep the branches clean.
